### ISSUE TITLE
[Word Online] Opening a document were and add-in was previously inserted causes context.sync() to fail

### ISSUE THREAD

**Issue by quefournier** [OfficeDev/office-js#4941]
This is related to [issue#4875](https://github.com/OfficeDev/office-js/issues/4875) as we were able to investigate further. 
Inserting an add-in into a document allows it to be used properly initially. However, the next time the document will be opened, every promises returned by context.sync() will remain pending forever.
 
## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web
* Host [Excel, Word, PowerPoint, etc.]: Word
* Office version number: N/A
* Operating System: Mac OS 14.5, Windows 11
* Browser (if using Office on the web): Chrome, Firefox & Edge
 
## Expected behavior

Our add-in makes use of .load() and other such methods to manipulate contentControls. Once Office.onReady() resolves correctly, we expect to be able to call office-js methods and update our document with context.sync().
 
## Current behavior

As mentioned in the summary above, if the add-in was already inserted when opening our document in Word Online, our add-in will appear to be in a "bugged state" where every call to context.sync() will never resolve.
 
We recently discovered that two iframes containing the add-in are inserted in the DOM once the add-in is loaded. Both frames are stacked, one above the other; only one of them is visible at a time (see pics 1, 2, 3 below, this exemple was made with a random add-in from the store). You can easily reproduce this issue with seemingly any add-in from the store. Select an add-in, insert it, and refresh your browser tab.
 
Alternatively, you can use the page we wrote and attached here (test.html).
 
A little note: inserting a manifest pointing to this page from a local file will not allow this bug to be reproduced. In our tests, we observed that this bug will only happen if the manifest is retrieved from the Office store or SharePoint hosted (in the appCatalog).
 
## Steps to reproduce

1. Set up a working environment where you can define a manifest that points to the attached web page (test.html). Remember that you can't insert the manifest from a local file or you probably won't be able to reproduce the issue.
2. Insert the add-in into your document. The script will try to access contentControls and log the result of the operation inside the page's body (cf picture 4).
3. Refresh your browser's tab.
4. The script should now fail (never-ending promise) and you should observe that two iframes exists in the dom (bearing the same id) (cf pictures 5&6)
 
## Link to live example(s)

We were not able to reproduce it in Script Lab.
 
## Context

We have two add-ins in production and this issue is preventing our client to use them properly.

# Provide additional details

This issue seems to never happen in Excel whether online or desktop, or Word desktop.

 
## Useful logs

- [ ] Console errors
- [X] Screenshots
- [ ] Test file (if only happens on a particular file)
 
Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

<img width="1107" alt="3" src="https://github.com/user-attachments/assets/217b715b-32e4-47a9-adfa-7761546c7032">
<img width="400" alt="3" src="https://github.com/user-attachments/assets/4121e867-975b-4c5d-88cb-7e3bcebb522d">
<img width="400" alt="3" src="https://github.com/user-attachments/assets/2c170674-3dfc-4258-8eb6-7340d7dcda06">

<img width="345" alt="4" src="https://github.com/user-attachments/assets/55494b2f-b405-4c0c-855d-3c43bd1e1bd2">
<img width="345" alt="5" src="https://github.com/user-attachments/assets/5ca144ca-1b51-49c5-aa45-fbc320d31685">
<img width="1920" alt="6" src="https://github.com/user-attachments/assets/aa9a50e8-bd77-4c62-83b9-6ac07d078d54">

[test.html.zip](https://github.com/user-attachments/files/17256339/test.html.zip)

**Comment by qinliuMSFT** [comment]
@quefournier thank you for reporting this issue, it has been put on our backlog #9414693, the product team will investigate and we will reply to you as soon as there is progress. Thank you for your patience.

**Comment by quefournier** [comment]
Hello, were you able to make any progress on this matter? Our clients are still stuck using Word Online. Thank you.

**Comment by xiruatms** [comment]
hi, @quefournier .  Thanks for reporting this. but I want to confirm further with you:

1. You mentioned that "You can easily reproduce this issue with seemingly any add-in from the store. Select an add-in, insert it, and refresh your browser tab.". What do you mean "any add-in" here? I tried, but seems I can't repro it.
2. How did you repro that with "Wikipedia"? can you help to take a screen record of the steps?

Detail information will hep us with further investigation.

**Comment by quefournier** [comment]
Hello @xiruatms. Thank you for investigating into this.
I tested it this morning and I admit this is not super easy to reproduce, is seems to be a bit random (except with our add-in of course, where it happens every time !)
To illustrate again wit wikipedia as an exemple (I still can make it work with I believe all add-ins) what I did this morning:

1. Load the add-in in a fresh document for the first time
2. Write some text in the word file (I just copy pasted random sentences)
3. Refresh the whole page (F5)
4. Strangely it didn't work the first few times. I had to refresh again a few more times.
5. Check the dom/sources to make sure the frame appears multiple times in the dom (see pictures in attachments)

![Capture d’écran 2024-11-25 à 11 04 02](https://github.com/user-attachments/assets/400261fa-987c-4f07-956e-504386b0a026)
![Capture d’écran 2024-11-25 à 11 04 29](https://github.com/user-attachments/assets/86b4f278-3ca5-4d23-a643-376006c5874f)

You can see that the frame appears 2 times in the dom. The top iframe is in a somewhat bugged state (every call to the office-js api will fail and the app will crash).

I'll be happy to make you a live demo if you have some time to spare.

**Comment by xiruatms** [comment]
@quefournier , are you still seeing this issue? We don't see this.

**Comment by quefournier** [comment]
@xiruatms Yes it's still happening and still a big issue to us. I'm willing to make a second demo if needed, as the above mention exemple is perhaps not the best.

**Comment by AmandaHan123** [comment]
Hi @quefournier , could you please provide the latest repro details to help us investigate?
